File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        sort(nums.begin(),nums.end());
     double maxx =DBL_MAX;
     int i=0;
     int j=nums.size()-1;
     while(i<j){
        double avg=(nums[i]+nums[j])/2.0;
         maxx=min(maxx,avg);
         i++;
         j--;
     }
     return maxx;
    }
};

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length - 1;
        double min_avg = Double.MAX_VALUE;
        for (int i = 0; i < nums.length / 2; i++) {
            double avg = (nums[i] + nums[n]) / 2.0;
            min_avg = Math.min(min_avg, avg);
            n--;
        }
        return min_avg;
    }
}